volume/store/store.go

```diff
@@ -309,16 +309,20 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
 // This makes sure there are no races between checking for the existence of a volume and adding a reference for it
 func (s *VolumeStore) GetWithRef(name, driverName, ref string) (volume.Volume, error) {
 	name = normaliseVolumeName(name)
```

Comment:
These two locks prevent different races, which you have analyzed in previous comments. If a lock would hold for a long time, such as 16s, we should catch the error to prevent it from happening again and again. This is why I catch the error at these 2 places. These 2 locks may take 16s until they unlock.

If the lock action is very quick, it is totally fine to ignore the error.